### ISSUE TITLE
Recent changes to conda-build-all broke conda-dev-recipes

### ISSUE THREAD

**Issue by jchodera** [omnia-md/conda-recipes#452]
@rmcgibbo, [`conda-dev-recipes`](https://github.com/omnia-md/conda-dev-recipes) is [now failing with](https://travis-ci.org/omnia-md/conda-dev-recipes/jobs/112179727#L6573-L6584):
```
Traceback (most recent call last):
  File "/io/conda-build-all", line 365, in <module>
    sys.exit(main())
  File "/io/conda-build-all", line 100, in main
    return execute(args, p)
  File "/io/conda-build-all", line 350, in execute
    code = build_package(m, python, numpy, args)
  File "/io/conda-build-all", line 309, in build_package
    force=args.force)
  File "/io/conda-build-all", line 225, in upload_to_binstar
    if is_prerelease(m):
NameError: name 'is_prerelease' is not defined
```
Looking at the code to see if I can figure out how to fix.

**Comment by jchodera** [comment]
I can't follow the logic here. The function `is_prerelease(m)` doesn't seem to be defined anywhere.

**Comment by jchodera** [comment]
Is this actually meant to be the function
```
def is_release_candidate(pkg_meta):                                                                                                                                                            
    from pkg_resources._vendor.packaging import version                                                                                                                                        
    parsed = version.parse(pkg_meta.version())                                                                                                                                                 
    return parsed.is_prerelease 
```

**Comment by jchodera** [comment]
Yeah, it must be this.

**Comment by jchodera** [comment]
Reopening issue because there is something else wrong even with the fix in #453.

**Comment by jchodera** [comment]
Travis didn't run, but this was the error on my machine:
```
[LSKI1497:omnia/conda-recipes/conda-recipes.jchodera] choderaj% ./conda-build-all * --check-against omnia
Traceback (most recent call last):
  File "./conda-build-all", line 365, in <module>
    sys.exit(main())
  File "./conda-build-all", line 100, in main
    return execute(args, p)
  File "./conda-build-all", line 328, in execute
    metadatas = sort_recipes(args_recipe)
  File "./conda-build-all", line 166, in sort_recipes
    return [name2m[name] for name in toposort(graph)]
  File "./conda-build-all", line 132, in toposort
    ":\n%s" % '\n'.join(repr(x) for x in data.items()))
ValueError: Cyclic dependencies exist among these items:
(u'pdbfixer', set([u'openmm']))
(u'sphinx_rtd_theme', set([u'sphinx']))
(u'ensembler', set([u'pdbfixer', u'openmm']))
(u'repex', set([u'openmmtools', u'openmm']))
(u'sphinxcontrib-autodoc_doxygen', set([u'sphinx']))
(u'sphinxcontrib-bibtex', set([u'sphinx']))
(u'omnia', set([u'pdbfixer', u'openmm']))
(u'parmed', set([u'openmm']))
(u'openmoltools', set([u'openmm', u'parmed']))
(u'alchemy', set([u'openmm']))
(u'openmmtools', set([u'openmm', u'parmed']))
(u'openmm', set([u'sphinxcontrib-bibtex', u'sphinx']))
(u'openmm-beta', set([u'sphinx', u'sphinxcontrib-autodoc_doxygen', u'sphinxcontrib-bibtex', u'sphinxcontrib-lunrsearch']))
(u'sphinxcontrib-lunrsearch', set([u'sphinx']))
(u'sphinx', set([u'sphinx_rtd_theme']))
(u'yank', set([u'openmmtools', u'openmoltools', u'openmm']))
```
